test/jdk/javax/swing/JMenuItem/4654927/bug4654927.java

```diff
@@ -48,34 +53,34 @@ public static void main(String[] args) throws Exception {
         try {
             String systemLAF = UIManager.getSystemLookAndFeelClassName();
             // the test is not applicable to Motif L&F
-            if(systemLAF.endsWith("MotifLookAndFeel")){
+            if (systemLAF.endsWith("MotifLookAndFeel")){
                 return;
             }
 
             UIManager.setLookAndFeel(systemLAF);
             Robot robot = new Robot();
-            robot.setAutoDelay(10);
+            robot.setAutoWaitForIdle(true);
+            robot.setAutoDelay(100);
 
-            SwingUtilities.invokeAndWait(new Runnable() {
+            SwingUtilities.invokeAndWait(() -> createAndShowUI());
 
-                public void run() {
-                    createAndShowUI();
-                }
-            });
             robot.waitForIdle();
+            robot.delay(1000);
 
             // test mouse press
             Point point = Util.getCenterPoint(menu);
             robot.mouseMove(point.x, point.y);
-            robot.mousePress(InputEvent.BUTTON1_MASK);
-            robot.mouseRelease(InputEvent.BUTTON1_MASK);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
             robot.waitForIdle();
+            robot.delay(250);
 
             point = Util.getCenterPoint(menuItem);
             robot.mouseMove(point.x, point.y);
-            robot.mousePress(InputEvent.BUTTON1_MASK);
-            robot.mouseRelease(InputEvent.BUTTON1_MASK);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
             robot.waitForIdle();
+            robot.delay(250);
```

Comment:
These don't help on the system I have tested on.
There's something else going on.
The test is basically about verifying that clicking on a disabled menu item doesn't dismiss the menu.
I don't think it does. If I comment out the press/release the same exception occurs.
And visually you can see the menu disappear and then reappear.
What it looks like is that the call at line 86/91 to mouseMove which returns the pointer to the "menu" location causes the list of menu items to disappear and then quickly reappear. Sampling if the menu is on screen at that time will cause the error.
It only happens once .. very odd ..

Reply:
Can you please tell which system it is not working?
 I guess harshita and vitaly tested on macos14 and it worked for them and I tested on other mac13,windows10 locally where it also worked, as well as in mach5 systems, whose link I have put in JBS.

I also have seen the menu disappear while clicking but it does not cause the test to fail..I saw the video recording vitaly uploaded in JBS and it is same as in macos13 where it pass always (we also dont see any failure for this test in CI runs so far)

Reply:
Add a delay to make it work on mach5 mac14 system. Tested for 50 iterations, link in JBS

Reply:
> I also have seen the menu disappear while clicking but it does not cause the test to fail..I saw the video recording vitaly uploaded in JBS and it is same as in macos13 where it pass always (we also dont see any failure for this test in CI runs so far)

But the menu should not disappear if a greyed out item is clicked. If you see it, you should address this problem.

Reply:
@prsadhuk 

> These don't help on the system I have tested on. There's something else going on. The test is basically about verifying that clicking on a disabled menu item doesn't dismiss the menu. I don't think it does. If I comment out the press/release the same exception occurs. And visually you can see the menu disappear and then reappear. What it looks like is that the call at line 86/91 to mouseMove which returns the pointer to the "menu" location causes the list of menu items to disappear and then quickly reappear. Sampling if the menu is on screen at that time will cause the error. It only happens once .. very odd ..

Ran the updated test MacOS 14 ARM, thought the test passes there seems to be an issue similar to what Phil described earlier - the menu appears then disappears and appears again quickly for the second scenario - when the menu is clicked and glided through the dropdown. I have attached the screen recording to the JBS.

Reply:
OK..But why the frame is at the right hand edge of the screen when we specify setLocationRelativeTo to center it?
Probably there is something wrong with setting location too in mac14 arm?

Reply:
I captured portion of the screen, hence it appears on the right edge in the recording.